### PR TITLE
fix: finds real base view if not explore name

### PR DESCRIPTION
**Pull request by noahamac**
Sometimes the base view is not the explore name. This happens most frequently when using the "from:" parameter,
https://docs.looker.com/reference/explore-params/from-for-explore
But can also happen after object extension such as for marketplace models. 

This changes the resolution from adding a properly named but empty table to using the first table in the explore's "sets" array, which after testing is consistent with desired functionality. 
Before:
image

After:
image

### CODE REVIEW

File: src/utils/LookmlDiagrammer/diagrammer.ts [hunk 1]
```diff
@@ -82,6 +82,7 @@ export function generateExploreDiagram(explore: ILookmlModelExplore, hiddenToggl
   const exploreJoins = explore.joins
   const fields = getFields(exploreFields)
   const views = getViews(exploreFields, exploreJoins, explore.name)
+  let baseViewName = ""
```
Comment (bryans99) [inline src/utils/LookmlDiagrammer/diagrammer.ts:85]: This use of let is okay :D

File: src/utils/LookmlDiagrammer/diagrammer.ts [hunk 2]
```diff
@@ -161,17 +166,17 @@ export function generateExploreDiagram(explore: ILookmlModelExplore, hiddenToggl
...
+  // General order tables would be arranged in, if no base view
   const joinCount = countJoins(diagramDict)
   let buildOrder = views.sort((a: string, b: string)=>{
     return joinCount[a] > joinCount[b] ? -1 : 1;
   })
 
   const diagrammer = new LookmlDiagrammer(diagramDict, buildOrder, explore)
   
-  const baseViewName = diagramDict.tableData[explore.name] ? explore.name : buildOrder[0]
+  const seedName = baseViewName !== "" ? baseViewName : buildOrder[0]
```
Comment (bryans99) [inline src/utils/LookmlDiagrammer/diagrammer.ts:177]: Is seedName the best name here? What's it used for?

**Review by bryans99 — APPROVED** [review]
LGTM although I'm not sure I like the choice of seedName as a a variable name

File: src/utils/LookmlDiagrammer/diagrammer.ts (hunk 2 above)
Comment (noahamac) [reply src/utils/LookmlDiagrammer/diagrammer.ts:177]: Will change. seedName is the table name to start arranging the diagram from

**Pull request merged**